cluster/validation.go

```diff
@@ -217,6 +217,12 @@ func validateNetworkOptions(c *Cluster) error {
 	if c.Network.Plugin == FlannelNetworkPlugin && c.Network.MTU != 0 {
 		return fmt.Errorf("Network plugin [%s] does not support configuring MTU", FlannelNetworkPlugin)
 	}
+
+	err := warnWeaveDeprecation(c.Version, c.Network.Plugin)
```

Comment:
check for `cniPlugin == WeaveNetworkPlugin` before calling so only version checks belong to the weave func itself

Reply:
I was checking this inside the function but I can do it outside too